src/main/java/address/controller/MainController.java

```diff
@@ -82,7 +82,7 @@ public MainController(MainApp mainApp, ModelManager modelManager, Config config)
     public void start(Stage primaryStage) {
         logger.info("Starting main controller.");
         this.primaryStage = primaryStage;
-        this.browserManager.initializeBrowser();
+        this.browserManager.start();
```

Comment:
I will have two method:
initBrowser() : to run init tasks in non-ui thread
start(): to run init tasks in ui thread

The start method in the MainController is in the ui thread.